tgui/packages/tgui/interfaces/MineralMagnet.js

```diff
@@ -0,0 +1,130 @@
+  const { act, data } = useBackend(context);
+
+  const { isLinked, magnetActive, magnetAutomaticMode, magnetCooldownOverride, magnetHealth, magnetLastUsed } = data;
+  const { time } = data;
+  const linkedMagnets = data.linkedMagnets || [];
+  const miningEncounters = data.miningEncounters || [];
+
+  const onCooldown = magnetLastUsed > time;
+
+  return (
+    <Window
+      width={300}
+      height={364}>
+      <Window.Content>
+        <Section title="Magnet Status">
+          <Box>
+            Condition: {isLinked ? getMagnetCondition(magnetHealth) : <Box inline color="bad">No Magnet Linked</Box>}
+          </Box>
+          <Box>
+            Status: {magnetActive ? "Pulling New Mineral Source" : (
+              onCooldown ? (
+                <>
+                  Cooling Down: <TimeDisplay value={Math.max(magnetLastUsed - time, 0)}
+                    timing
+                    format={value => formatTime(value)} />
+                </>
+              ) : "Idle"
+            )}
+          </Box>
+        </Section>
+        {isLinked ? (
+          <Section title="Magnet Controls">
+            <Button
+              textAlign="center"
+              color="teal"
+              icon="rss"
+              onClick={() => act('geoscan')}
+              fluid >
+              Scan
+            </Button>
+            <Section textAlign="center">
+              {(!!magnetActive || (onCooldown && !magnetCooldownOverride)) && (
+                <Dimmer fontSize={1.75} pb={2}>
+                  {magnetActive ? "Magnet Active" : "On Cooldown"}
+                </Dimmer>
+              )}
+              <Button
+                textAlign="center"
+                color={onCooldown && magnetCooldownOverride ? "average" : "purple"}
+                icon="magnet"
+                onClick={() => act('activatemagnet')}
+                fluid >
+                Activate Magnet
+              </Button>
+              <Collapsible title={<><Icon name="search" />Activate telescope location</>}
+                textAlign="center"
+                color={onCooldown && magnetCooldownOverride ? "average" : "purple"} >
+                <Section height={7} pl={2} scrollable fill>
+                  {miningEncounters.map(encounter => (
+                    <Button key={encounter.id}
+                      onClick={() => act('activateselectable', { encounter_id: encounter.id })}
+                      fluid>
+                      {encounter.name}
+                    </Button>
+                  ))}
+                </Section>
+              </Collapsible>
+              <Button.Checkbox checked={magnetCooldownOverride}
+                onClick={() => act('overridecooldown')}
+                style={{ 'z-index': '2' }}>
+                Override Cooldown
+              </Button.Checkbox>
+              <Button.Checkbox checked={magnetAutomaticMode}
+                onClick={() => act('automode')}
+                style={{ 'z-index': '2' }}>
```

Comment:
Why is z-index needed? I suspect it's not (as I believe it's added in React via the `style` prop with the key `zIndex`).